pandas/core/groupby/generic.py

```diff
@@ -134,8 +143,23 @@ class NamedAgg(NamedTuple):
     2           1      12.0
     """
 
-    column: Hashable
-    aggfunc: AggScalar
+    def __new__(cls, column, aggfunc, *args, **kwargs):
+        if not isinstance(aggfunc, str):
+            aggfunc = cls._get_wrapped_aggfunc(aggfunc, *args, **kwargs)
+        self = _BaseNamedAgg.__new__(cls, column, aggfunc)
+        return self
+
+    @staticmethod
+    def _get_wrapped_aggfunc(function, *initial_args, **initial_kwargs):
+        def wrapped_aggfunc(*new_args, **new_kwargs):
+            final_args = new_args + initial_args
+            final_kwargs = {**initial_kwargs, **new_kwargs}
+            return function(*final_args, **final_kwargs)
+
+        return wrapped_aggfunc
+
+    def __repr__(self):
```

Comment:
Is this related to the args/kwargs forwarding?

Reply:
The __repr__ is unnecessary, it related to a previous solution that I did not use. I will rollback this change.